### PR TITLE
XCODE support for v5.2.3 of the FIPS module

### PR DESCRIPTION
**Pull request by kaleb-himes**
# Description

Fixes some warnings found by XCODE and adds options for customer(s) who will be depending on v5.3.0 of the FIPS module.

# Testing

See fips repo for test applications, full harness and optesting done in-house, ready to go to a CSTL for review

# Checklist

 - [X] added tests
 - [ ] updated/added doxygen
 - [X] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

**Review by lealem47 — DISMISSED** [review]

**Comment by lealem47** [comment]
Jenkins Retest this please

**Comment by kaleb-himes** [comment]
NOTE: After internal versioning discussion, changing version to v5.2.3 from v5.3.0 which is in contention with fips-ready and fips-dev

File: IDE/XCODE-FIPSv5/user_settings.h [hunk 1]
```diff
@@ -374,6 +432,7 @@ extern "C" {
 #undef WOLFSSL_SHA3
 #if 1
     #define WOLFSSL_SHA3
+    #define Sha3 wc_Sha3
```
Comment (JacobBarthelmeh) [inline IDE/XCODE-FIPSv5/user_settings.h:435]: This should be handled in wolfssl/wolfcrypt/sha3.h as long as NO_OLD_WC_NAMES is not defined?

**Comment by JacobBarthelmeh** [comment]
This looks good, @kaleb-himes please rebase it onto master branch when you get a chance and resolve the merge conflict.

File: IDE/XCODE-FIPSv5/user_settings.h (hunk 1 above)
Comment (kaleb-himes) [reply IDE/XCODE-FIPSv5/user_settings.h:435]: sha3.h gets replaced by an older version of sha3.h when FIPS is checked out, unfortunately that older version doesn't handle this.

Comment (kaleb-himes) [reply IDE/XCODE-FIPSv5/user_settings.h:435]: Double checking to be sure.

Comment (kaleb-himes) [reply IDE/XCODE-FIPSv5/user_settings.h]: Ope, I was wrong. Good catch, thank you! Updated user_settings.h to define `NO_OLD_WC_NAMES`.

**Comment by kaleb-himes** [comment]
rebased on Jun 21 2024

**Review by JacobBarthelmeh — APPROVED** [review]

**Pull request merged**